mirrord-layer/src/lib.rs

```diff
@@ -97,116 +98,175 @@ fn init() {
-                        let stream = match TcpStream::connect(addr).await {
-                            Ok(stream) => stream,
-                            Err(err) => {
-                                error!("failed to connect to port {:?}", err);
-                                continue;
-                            }
-                        };
-                        let (sender, receiver) = channel::<TcpTunnelMessages>(1000);
-                        active_connections.insert(conn.connection_id, sender);
-                        task::spawn(async move {
-                            tcp_tunnel(stream, receiver).await
-                        });
-                    }
-                    Some(Ok(DaemonMessage::TCPData(msg))) => {
-                        let sender = match active_connections.get(&msg.connection_id) {
-                            Some(sender) => sender,
-                            None => {
-                                debug!("no sender for {:?}", msg.connection_id);
-                                continue;
-                            }
-                        };
-                        if let Err(err) = sender.send(TcpTunnelMessages::Data(msg.data)).await {
-                                debug!("sender error {:?}", err);
-                                active_connections.remove(&msg.connection_id);
-                        }
-                    },
-                    Some(Ok(DaemonMessage::TCPClose(msg))) => {
-                        let sender = match active_connections.remove(&msg.connection_id) {
-                            Some(sender) => sender,
-                            None => {
-                                debug!("no sender for {:?}", msg.connection_id);
-                                continue;
-                            }
-                        };
-                        if let Err(err) = sender.send(TcpTunnelMessages::Close).await {
-                                debug!("sender error {:?}", err);
-                        }
-                    }
-                    Some(_) => {
-                        debug!("NONE in some");
-                        break
-                    },
-                    None => {
-                        thread::sleep(Duration::from_millis(2000));
-                        debug!("NONE in none");
-                        continue
-                    }
-                }
+            daemon_message = codec.next() => {
+                handle_daemon_message(&open_file_handler, &mut port_mapping, &mut active_connections, daemon_message.unwrap().unwrap()).await;
             }
         }
     }
 }
 
+/// Enables file and socket hooks.
 fn enable_hooks() {
-    let interceptor = Interceptor::obtain(&GUM);
-    sockets::enable_hooks(interceptor)
+    let mut interceptor = Interceptor::obtain(&GUM);
+    interceptor.begin_transaction();
+
+    sockets::enable_socket_hooks(&mut interceptor);
+    file::enable_file_hooks(&mut interceptor);
```

Comment:
Make file hooks optional by env var, to be confirmed with @eyalb181 (version wise and UX wise it'd be weird if a new version starts hooking file operations without a flag set, if we don't release it as major)